### PR TITLE
AST-25406 Add gl-sast report

### PR DESCRIPTION
**Pull request by margaritalm**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: internal/commands/result.go [hunk 1]
```diff
@@ -1221,6 +1247,85 @@ func convertCxResultsToSarif(results *wrappers.ScanResultsCollection) *wrappers.
...
+	glSast.Vulnerabilities = glVulnra
+	return glSast
+}
+
+func convertCxResultToGlVulnerability(results *wrappers.ScanResultsCollection, glSast *wrappers.GlSastResultsCollection) []wrappers.GlVulnerabilities {
+	for _, result := range results.Results {
+		engineType := strings.TrimSpace(result.Type)
+		if engineType == commonParams.SastType {
+			glSast = parseGlSastVulnerability(result, glSast)
+		} /*else if engineType == commonParams.KicsType {
```
[the remark this answers is not in the document]
Comment (margaritalm) [reply internal/commands/result.go:1264]: Removed

File: internal/commands/result.go [hunk 2]
```diff
@@ -851,6 +852,10 @@ func createReport(format,
 		jsonRpt := createTargetName(targetFile, targetPath, printer.FormatJSON)
 		return exportJSONResults(jsonRpt, results)
 	}
+	if printer.IsFormat(format, printer.FormatGL) {
+		jsonRpt := createTargetName("gl-sast-report", targetPath, "json")
```
[the remark this answers is not in the document]
Comment (margaritalm) [reply internal/commands/result.go:856]: Done

File: internal/commands/result.go [hunk 3]
```diff
@@ -988,7 +993,28 @@ func exportSarifResults(targetFile string, results *wrappers.ScanResultsCollecti
 	_ = f.Close()
 	return nil
 }
-
+func exportGlSastResults(targetFile string, results *wrappers.ScanResultsCollection, summary *wrappers.ResultSummary) error {
+	var err error
+	var resultsJSON []byte
+	log.Println("Creating gl-sast Report: ", targetFile)
+	var glSastResults = convertCxResultsToGLSast(results)
```
[the remark this answers is not in the document]
Comment (margaritalm) [reply internal/commands/result.go:1000]: Done

File: internal/commands/result.go [hunk 4]
```diff
@@ -1221,6 +1247,85 @@ func convertCxResultsToSarif(results *wrappers.ScanResultsCollection) *wrappers.
...
+		},
+		Location: wrappers.Location{
+			File:      fileName,
+			StartLine: startLine,
+			EndLine:   endLine,
+			Class:     fileName,
+		},
+	})
+	return glSast
+}
```
[the remark this answers is not in the document]
Comment (margaritalm) [reply internal/commands/result.go:1316]: Done

File: internal/commands/result.go [hunk 5]
```diff
@@ -1221,6 +1248,85 @@ func convertCxResultsToSarif(results *wrappers.ScanResultsCollection) *wrappers.
...
+	ID := fmt.Sprintf("%s:%s:%s", queryName, fileName, lineNumber)
+	category := fmt.Sprintf("%s-%s", wrappers.VendorName, result.Type)
+	message := fmt.Sprintf("%s@%s:%s", queryName, fileName, lineNumber)
+
+	glSast.Vulnerabilities = append(glSast.Vulnerabilities, wrappers.GlVulnerabilities{
+		ID:       ID,
+		Category: category,
+		Name:     queryName,
+		Message:  message,
+		//	Message:     queryName + "@" + fileName + " : " + lineNumber,
```
Comment (igorlombacx) [inline internal/commands/result.go:1285]:
```suggestion
```

**Review by igorlombacx — CHANGES_REQUESTED** [review]

**Review by igorlombacx — APPROVED** [review]
Great job! LGTM!

**Pull request merged**